store/generated_key/v2/eigenda.go

```diff
@@ -141,12 +167,76 @@ func (e Store) BackendType() common.BackendType {
 //
 // Since v2 methods for fetching a payload are responsible for verifying the received bytes against the certificate,
 // this Verify method only needs to check the cert on chain. That is why the third parameter is ignored.
-func (e Store) Verify(ctx context.Context, certBytes []byte, _ []byte) error {
+//
+// TODO: this whole function should be upstreamed to a new eigenda VerifyingPayloadRetrieval client
```

Comment:
Are we sure we want to put the recency check in that client? I can't remember if we came to a conclusion on that yet

Reply:
What's your argument for not adding it to the client? After what bowen said last week that the "CertVerifier" should really be called a "DataAvailabilityVerifier" because its whole purpose is to parse a cert and make sure that if we try to download a blob, that blob should be available. And the recencyCheck is a big part of that (we could said it super laxly to a 14 day window for example for non-rollups). Its also not used by default (if the WindowSize is not set, it defaults to 0 which skips the check), so I don't personally see a reason why not to upstream it. But curious why you think otherwise

Reply:
> But curious why you think otherwise

I didn't have a strong opinion one way or the other, just wanted to make sure this is the way to go.

Your arguments are persuasive, I'm on board with the plan of putting this into the client.